packages/ceramic-core/src/document.ts

```diff
@@ -72,7 +73,9 @@ class Document extends EventEmitter {
       validate = true,
   ): Promise<Document> {
     const genesis = await dispatcher.retrieveRecord(genesisCid)
-    const id = DoctypeUtils.createDocIdFromGenesis(genesisCid)
+  
+    const doctype = (await context.ipfs.dag.get(genesis.link)).value.doctype
+    const id = new DocID(doctype, genesisCid)
```

Comment:
Should we not change the function signature of this function to get a DocID instead of genesisCID?

Reply:
hmm yeah i could could go either way here, seems find for create to be made from parts, but maybe makes most sense if docid created in func where both type and genesis cid are easily available, and docid passed here

Reply:
I think we eventually want to remove the doctype property on the genesis record. So this would be required.